src/common/snippets/src/lowered/pass/validate.cpp

```diff
@@ -0,0 +1,184 @@
+        const auto& shape = desc->get_shape();
+        const auto& layout = desc->get_layout();
+        const auto max_dim = *std::max_element(layout.begin(), layout.end());
+        OPENVINO_ASSERT(max_dim < shape.size(), "Max layout index can't be larger than the shape size");
+        OPENVINO_ASSERT(shape.size() == layout.size(), "Shape and layout must have the same length");
+    };
+    const auto& in_descs = expr->get_input_port_descriptors();
+    const auto& out_descs = expr->get_output_port_descriptors();
+    std::for_each(in_descs.cbegin(), in_descs.cend(), validate_descriptor);
+    std::for_each(out_descs.cbegin(), out_descs.cend(), validate_descriptor);
+}
+
+void validate_parameter(const ExpressionPtr& expr, const LinearIR& linear_ir) {
+    OPENVINO_ASSERT(ov::is_type<ov::op::v0::Parameter>(expr->get_node()),
+                    "Parameter validation expects Parameter op");
+    auto consumer_inputs = expr->get_output_port_connector(0)->get_consumers();
+    const auto& first_consumer = consumer_inputs.begin()->get_expr();
+    if (is_type<snippets::op::RankNormalization>(first_consumer->get_node())) {
+        OPENVINO_ASSERT(consumer_inputs.size() == 1,
+                        "If there is RankNormalization after Parameter, it should be single consumer of the Parameter");
+        consumer_inputs = first_consumer->get_output_port_connector(0)->get_consumers();
+    }
+    std::set<std::vector<size_t>> layouts;
+    for (const auto& consumer_input : consumer_inputs) {
+        const auto& node = consumer_input.get_expr()->get_node();
+        if (const auto ma = ov::as_type_ptr<snippets::op::MemoryAccess>(node)) {
+            OPENVINO_ASSERT(ma->is_memory_access_input_port(consumer_input.get_index()),
+                            "Parameter expects MemoryAccess on output");
+            layouts.insert(consumer_input.get_descriptor_ptr()->get_layout());
+        } else {
+            OPENVINO_ASSERT(ov::is_type<op::LoopEnd>(node), "Parameter must be connected to MemoryAccess op or LoopEnd");
+        }
+    }
+    OPENVINO_ASSERT(layouts.size() == 1, "All consumers of Parameter must have the same layout");
+}
+
+void validate_result(const ExpressionPtr& expr, const LinearIR& linear_ir) {
+    OPENVINO_ASSERT(ov::is_type<ov::op::v0::Result>(expr->get_node()),
+                    "Result validation expects Result op");
+    const auto source = expr->get_input_port_connector(0)->get_source();
+    const auto ma = ov::as_type_ptr<snippets::op::MemoryAccess>(source.get_expr()->get_node());
+    OPENVINO_ASSERT(ma && ma->is_memory_access_output_port(source.get_index()),
+                    "Result expects MemoryAccess parent");
+}
+
+void validate_buffer(const ExpressionPtr& expr, const LinearIR& linear_ir) {
+    OPENVINO_ASSERT(ov::is_type<op::Buffer>(expr->get_node()),
+                    "Buffer validation expects Buffer op");
+    for (const auto& in : expr->get_input_port_connectors()) {
```

Comment:
Is it a valid configuration if a Buffer has several inputs?
Sounds like an inplace Concat or smth

Reply:
At the moment, `Buffer` support only single input and output. This code was coped from `Parameter` and `Result`. Let's cover the real case for now.  Corrected. Thanks!